pykickstart/commands/rhsm.py

```diff
@@ -96,9 +96,9 @@ def _getParser(self):
                         ``--proxy=[protocol://][username[:password]@]host[:port]``
                         """)
         op.add_argument("--server-hostname", metavar="<subscription_service_server_url>", version=RHEL8, required=False,
-                        help="Red Hat subscription service server URL.")
+                        help="Red Hat Satellite server URL (registration to Satellite is supported since RHEL 9.0).")
         op.add_argument("--rhsm-baseurl", metavar="<content_base_url>", version=RHEL8, required=False,
-                        help="Content base URL.")
+                        help="Content base URL. Set automatically when registering to Red Hat Satellite.")
```

Comment:
So this argument really is the content base url? With RHSM I'd expect it to be Candlepin, not Pulp.

When `--server-hostname` is given, is the argument still respected or actually ignored?

Reply:
In RHEL 8, `--server-hostname` just sets the `hostname` key in the server section of the `rhsm.conf` via DBus. The `--rhsm-baseurl` just sets the `baseurl` key in the `rhsm` section of the config.

In RHEL 9 we still do the same - if --server-hostname or --rhsm-baseurl are present, we set the respective keys in the config file. But also do one extra thing - if --server-hostname is set we consider it to be a request to register to Satellite on the given URL: https://github.com/rhinstaller/anaconda/blob/rhel-9/pyanaconda/modules/subscription/runtime.py#L1025

So we do fetch the Satellite provisioning script and execute it + some other minor things.

The Satellite provisioning script usually sets the `baseurl` key in the `rhsm` section of `rhsm.conf` when executed, then indeed, the value provided to `--rhsm-baseurl` can be most likely considered as effectively ignored. 

This is because IIRC the values are first set in the RHSM config & then the script is executed, overwriting the baseurl:
https://github.com/rhinstaller/anaconda/blob/rhel-9/pyanaconda/ui/lib/subscription.py#L236
https://github.com/rhinstaller/anaconda/blob/rhel-9/pyanaconda/ui/lib/subscription.py#L279

Reply:
That helps a lot. Thanks!